docs/monitoring-tech.md

```diff
@@ -0,0 +1,193 @@
+# Import System Documentation
+
+## Purpose
+
+The import system automatically collects ZIP files from an external SFTP source (e.g., IT-AMS), processes contained XML files, maps them to internal database models, stores them, and optionally runs post-processing (like creating Monitoring Goals).
+
+---
+
+## High-Level Flow
+
+1. **Scheduled Execution** via cron or manual trigger.
+
+2. **Download Phase**
+
+   - Connects to SFTP using stored credentials
+   - Lists & filters available files
+   - Downloads a file and uploads it to S3
+   - Tracks attempts, retries on failure
+
+3. **Processing Phase**
+
+   - Downloads the ZIP file from S3
+   - Extracts required XML files based on definitions
+   - Parses, maps, and stores data into corresponding DB tables
+   - Soft-deletes missing records
+
+4. **Post-Processing Phase** (optional)
+
+   - Executes tasks like creating Monitoring Goals
+
+---
+
+## Code Modules
+
+### `index.ts`
+
+Exports orchestrator functions:
+
+- `download(importId, timeBox)`
+- `process(importId)`
+- `moreToDownload(importId)`
+- `moreToProcess(importId)`
+- `getImportSchedules()`
+
+### `download.ts`
+
+- Collects available files from SFTP
+- Uploads to S3 (with hash)
+- Tracks collection attempts
+- Skips if time-boxed
+
+### `processZipFileFromS3.ts`
+
+- Pulls file from S3
+- Uses `ZipStream` to extract contents
+- Records available XMLs
+- Calls `processFilesFromZip.ts`
+
+### `processFilesFromZip.ts`
+
+- Iterates through process definitions
+- Uses `zipClient` to pull stream
+- Passes to `processFile.ts`
+
+### `processFile.ts`
+
+- Pipes file stream through Hasher + Encoding Converter
+- Initializes XML parser
+- Calls `processRecords.ts`
+
+### `processRecords.ts`
+
+- Remaps raw XML
+- Filters for DB model columns
+- Based on keys:
+  - Insert new
+  - Update existing
+  - Mark missing as soft-deleted
+
+### `record.ts`
+
+Handles:
+
+- File status & metadata tracking
+- Insertion & update of ImportFile / ImportDataFile
+- Determines prior file for delta logic
+
+### `postProcess.ts`
+
+Runs post-processing functions defined in `Import.postProcessingActions`. Example:
+
+- `createMonitoringGoals`
+
+---
+
+## Sample Process Definition JSON
+
+```json
+{
+  "keys": ["reviewId"],
+  "path": ".",
+  "encoding": "utf16le",
+  "fileName": "Review.xml",
+  "remapDef": {
+    ".": "toHash.*",
+    "EndDate": "endDate",
+    "Outcome": "outcome",
+    "ReviewId": "reviewId",
+    "StatusId": "statusId",
+    "ContentId": "contentId",
+    "StartDate": "startDate",
+    "ReviewType": "reviewType",
+    "ReportDeliveryDate": "reportDeliveryDate"
```

Comment:
the following as well, I think?

```json
{
  "ReportAttachmentId": "reportAttachmentId",
  "Name": "name"
}
```

from https://github.com/HHS/Head-Start-TTADP/blob/main/src/migrations/20240509221012-monitoring-findings.js#L828-L848

Reply:
There are multiple definitions of this file. I'll create a ticket to clean up.

Reply:
I adjusted the documentation to use the updated version with report attachment id